### ISSUE TITLE
Check magnetic table readout

### ISSUE THREAD

**Issue by CarlOlsson** [PX4/PX4-SITL_gazebo-classic#321]
There was[ a bug in ECL](https://github.com/PX4/ecl/pull/606) regarding the readout of the magnetic tables sometimes resulting in a large error in magnetic declination and inclination. We need to check that the same error is not present in SITL

**Comment by Jaeyoung-Lim** [comment]
@CarlOlsson Any updates regarding this issue?

**Comment by CarlOlsson** [comment]
No, I think we should use the same library for calculating the magnetic field in ECL and SITL. I.e. moving the ecl geo functionality to its own submodule and also include that in SITL would probably be the best

**Comment by dagar** [comment]
It's perhaps a bit convoluted, but you could include ecl in sitl_gazebo and use only the ecl_geo library.

**Comment by CarlOlsson** [comment]
either that or include ecl_geo in ecl and sitl_gazebo, which one do you prefer?

**Comment by TSC21** [comment]
I do prefer the approach of bringing `ecl_geo` to its own submodule and probably rename it, so it becomes agnostic for both `sitl_gazebo` and `ecl`.

**Comment by LorenzMeier** [comment]
@Jaeyoung-Lim Could you have a look?
